src/gui/utils/LevelMeterComponent.cpp

```diff
@@ -43,24 +50,22 @@ void LevelMeterComponent::paint (Graphics& g)
     const auto yPad = meterBounds.getY() / 2;
     g.fillRect (leftChBounds.withTop (getYForDB (dbLevels[0]) + yPad));
     g.fillRect (rightChBounds.withTop (getYForDB (dbLevels[1]) + yPad));
-
-    //    g.setFont (12.0f);
-    //    g.setColour (Colours::black);
-    //    for (auto dbLevel : { -45.0f, -30.0f, -15.0f, 0.0f })
-    //    {
-    //        auto y = (float) getYForDB (dbLevel);
-    //        auto right = (float) meterMarkBounds.getRight();
-    //        g.drawLine (right - 6.0f, y, right, y);
-    //
-    //        const auto labelBounds = meterMarkBounds.withHeight (20).withCentre (juce::Point { meterMarkBounds.getCentreX(), (int) y });
-    //        g.drawFittedText (String ((int) dbLevel), labelBounds, Justification::centred, 1);
-    //    }
 }
 
 void LevelMeterComponent::timerCallback()
 {
+    bool needsRepaint = false;
     for (int ch = 0; ch < 2; ++ch)
+    {
         dbLevels[ch] = Decibels::gainToDecibels (levelDetector[ch].processSample (rmsLevels[ch]));
 
-    repaint();
+        if (std::abs (dbLevels[ch] - dbLevelsPrev[ch]) > 0.5f && dbLevels[ch] > minDB && dbLevelsPrev[ch] > minDB)
```

Comment:
I liked your pattern for this, so I figured I'd try something similar with the level meters. I'm trying a threshold of 0.5 dB, but I'm curious if you think it looks a tiny bit choppy?

Reply:
Yeah It might be a bit choppier than it was before? I don't necessarily think thats a bad thing if it's adding efficiency? The input and output meters to me are more of a "see if a signal is coming though or not" so as long as they show that, to me thats all I would look for.

Reply:
Yeah, that's true... maybe lowering the threshold just a touch wouldn't hurt too much though. It's also a bit hard to tell atm, since everything looks a little bit choppy when `JUCE_ENABLE_REPAINT_DEBUGGING` is turned on :).

Reply:
Yeah good point, `JUCE_ENABLE_REPAINT_DEBUGGING` is not a good way to judge UI's

Reply:
I tried with `JUCE_ENABLE_REPAINT_DEBUGGING` turned off and it looked okay to me, so hopefully that's fine!